client/server/silicon/silicon_utils.ts

```diff
@@ -0,0 +1,21 @@
+import { EncryptedMarket, Market } from '../../graphql/graphql';
+import { Silicon } from './silicon'; // Assume this is the path to your Silicon class
+import { poseidonHashMany } from "@scure/starknet";
+
+export function verifySiliconMapping(encrypted_silicon: Silicon<EncryptedMarket>, transparent_silicon: Silicon<Market>): boolean {
```

Comment:
BlindedMarket may be more appropriate since vanilla poseidon isn't technically an encryption algorithm (encryption implies a two way function)

It's a nit, but important since we're a cryptography focused company :)